fs/fs_zenfs.cc

```diff
@@ -1459,9 +1457,17 @@ IOStatus ZenFS::MigrateFileExtents(
 extern "C" FactoryFunc<FileSystem> zenfs_filesystem_reg;
 
 FactoryFunc<FileSystem> zenfs_filesystem_reg =
+#ifdef ZENFS_USE_OLD_REGISTRATION_API
```

Comment:
```suggestion
if (ROCKSDB_MAJOR < 7) {...
```
I think it would be more user friendly to use the defined version macros. Using conditional c expression instead of conditional compile should be ok, the compiler will strip the dead code.

Reply:
Uhm, how is this supposed to work? How could we conditionally define  the registration in runtime?(!)

Reply:
But hey, i just learned that we CAN get the preprocessor to do it.
```


#if ROCKSDB_MAJOR < 7
..
# else
..
#endif
```

Reply:
problem is that we're not at 7 yet, maybe we can compare with 6.29.0

Reply:
> If you get that to compile.. :)

Well, add the rest of the conditional brackets, etc. why would it not compile?

Reply:
> problem is that we're not at 7 yet, maybe we can compare with 6.29.0

Problem with that approach is that FB may release a version 6.29.0 with the old API, and then we break. They should probably have updated the major when they did the breaking change :shrug: We could raise an issue in rocksdb tracker to ask about that.

Reply:
@metaspace : I just pushed an update, i think the latest version is safe, i checked pre/post breaking change and it seems ok